packages/framework-provider-local/src/library/web-socket-server-adapter.ts

```diff
@@ -0,0 +1,110 @@
+import { Server, WebSocket } from 'ws'
+import { Server as HttpServer } from 'http'
+import { Duplex } from 'stream'
+import { BoosterConfig, EventType, UUID } from '@boostercloud/framework-types'
+import { IncomingMessage } from 'node:http'
+import { GraphQLService } from '../services'
+import * as express from 'express'
+import { getLogger } from '@boostercloud/framework-common-helpers'
+
+export interface ConnectionContext {
+  connectionId: UUID
+  eventType: EventType
+}
+
+export interface ExpressWebSocketMessage {
+  connectionContext: ConnectionContext
+  incomingMessage?: IncomingMessage
+  data?: unknown
+}
+
+const LOCAL_ENVIRONMENT_WEBSOCKET_SERVER_PORT = 'LOCAL_ENVIRONMENT_WEBSOCKET_SERVER_PORT'
+
+export class WebSocketServerAdapter {
+  private readonly httpServer: HttpServer
+  private readonly _webSocketServer: Server
+  private readonly clients: Record<string, WebSocket> = {}
+  private readonly WS_PORT = process.env[LOCAL_ENVIRONMENT_WEBSOCKET_SERVER_PORT] ?? '65529'
+
+  constructor(readonly graphQLService: GraphQLService, config: BoosterConfig) {
+    const expressServer = express()
+    this._webSocketServer = new WebSocket.Server({
+      noServer: true,
+      path: '/websockets',
```

Comment:
Nitpicking: This should be singular, as it is the only websocket we create
```suggestion
      path: '/websocket',
```